flaskext/htmlbuilder.py

```diff
@@ -122,92 +129,96 @@ def render(element, level=0):
         </form>
 
     The :func:`render` function accepts the following arguments:
-    
+
     :param element: element created through the `html` builder instance, a list
                     of such elements, or just a string.
     :param level: indentation level of the rendered element with a step of two
                   spaces.  If `level` is `None`, the `element` will be rendered
-                  without indentation and new line transferring, passing the 
+                  without indentation and new line transferring, passing the
                   same rule to the child elements.
     """
+
     if hasattr(element, 'render'):
         return element.render(level)
-    elif hasattr(element, '__iter__'):
-        return _render_iteratable(element, level)
     elif isinstance(element, basestring):
         return _render_string(element, level)
+    elif hasattr(element, '__iter__'):
+        return _render_iteratable(element, level)
     elif element is None:
         return ''
-    
+
     raise TypeError('Cannot render %r' % element)
 
 
 class BaseElement(object):
     __slots__ = []
-    
+
     def __str__(self):
         return str(self.render(None))
-    
+
     def __unicode__(self):
         return unicode(self.render(None))
-    
+
     def __html__(self):
         return self.__unicode__()
-    
+
     def render(self, level):
         raise NotImplementedError('render() method has not been implemented')
 
 
 class Element(BaseElement):
     __slots__ = ['_name', '_children', '_attributes']
-    
+
     def __init__(self, name):
         self._name = _unmangle_element_name(name)
-        
+
         # `None` indicates a void element or a list content for non-void
-        # elements. 
+        # elements.
         self._children = None
-        
+
         # `None` indicates no attributes, or it is a list if there are any.
         self._attributes = None
-        
+
     def __call__(self, *children, **attributes):
         # Consequent calling the instances of that class with keyword
         # or list arguments or without arguments populates the HTML element
         # with attribute and children data.
-        
+
         if attributes:
             # Keyword arguments are used to indicate attribute definition.
-            self._attributes = attributes
+            self._attributes = collections.OrderedDict(sorted(
```

Comment:
kwargs doesn't keeps the order then the expected string [sometimes] will be different than the result (I mean, the order of the attributes can be different)